File: reference/springboot/domain-usecases/src/main/java/org/example/domain/usecases/types/AnimalVisitor.java
package org.example.domain.usecases.types;

/*
  oneOf:
    - $ref: "#/dog"
    - $ref: "#/cat"
    - $ref: "#/cow"
 */
public interface AnimalVisitor<T> {

    T visit(Dog dog);

    T visit(Cat cat);

    T visit(Cow cow);
}

File: reference/dotnet/Domain/Company.Product.Domain.UseCases/Types/AnimalVisitor.cs
namespace Company.Product.Domain.UseCases.Types
{
    /*
    oneOf:
        - $ref: "#/dog"
        - $ref: "#/cat"
        - $ref: "#/cow"
    */
    public interface AnimalVisitor<T>
    {
        T Visit(Cat cat);
        T Visit(Cow cow);
        T Visit(Dog dog);
    }
}